app/src/main/java/ch/epfl/qedit/model/Server.java

```diff
@@ -0,0 +1,115 @@
+package ch.epfl.qedit.model;
+
+import androidx.annotation.NonNull;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class Server {
+    private boolean isSignInSuccessful = false;
```

Comment:
Why exactly do we need `isCreateSuccessful` and `isSignInSuccessful`, since the methods `signIn` and `createUser` already return a boolean flag ?

Reply:
Because it is `task.isSuccessful()` that will say if either `signIn` or `createUser` will be successful and we need to access it outside of the innerClass, so I created those Boolean variables `isCreateSuccessful` and `isSignInSuccessful`.

But I just realized that I can delete them and use the method `.isSuccessful()` on both `createUserWithEmailAndPassword` and `signInWithEmailAndPassword`. I could delete a few methods as well. I think I will keep the `onComplete` method as well. It could be useful maybe later.

Reply:
But I think for now I will do it this way and for later versions I will study how to get the error codes to have something better in the long run.